coordinator/scripts/luigi_scoop_server.py

```diff
@@ -0,0 +1,206 @@
+      rospy.loginfo('%s: Preempted' % self._action_name)
+      self.server_.set_preempted()
+      success = False
+
+    # Go to the start point, Arne should give this point from simulation
+    startPoint = Point(x=-0.1, y=-0.4, z=0.25)
+
+    # TODO: Check the successs of the return
+    scoopingResponse = self.TranslationalPTPMotionClient(startPoint, startPoint)
+
+    # success of the total scoping operation
+    success = True
+    # We try to scoop in just five points sequence
+    for scoop in range(len(scoops)):
+
+      # Go to the start point of the box, depending on the flavor
+      self.GoToFlavor(flavors[scoop])
+
+      # Loop for how many scoops of this flavor, really luigi??
+      for scoopPerFlavor in range(scoops[scoop]):
+        
+        # Each scoop is done in scoopingSteps
+        for i in range(scoopingSteps):
+          # Wait for scooping_planning/status to be idle
+          # Calculate new point
+          # Send point to scooping_planning/scoop
+
+          # Wait for scooping_planning/status to be idle
+          while not self.scooping_status_ == 'IDLE':
+            self.scooping_human_status_ = self.scooping_status_ + ' need more time'
+            rospy.sleep(1.)
+
+          # If not IDLE, update scooping status
+          self.scooping_human_status_ = self.scooping_status_ + ' in step ' + str(i) + ' out of 5'
+
+
+          # Now for step 2 and 3
+          # Move along the y-z plane (ice cream surface)     
+          scoopingResponse = self.TranslationalPTPMotionClient(Point(x=-0.1, y=-0.4, z=0.25), Point(x=-0.1, y=-0.3, z=0.25))
+          if not scoopingResponse:
+            self.scooping_human_status_ = 'Failed to come up with a plan for step ' + str(i) + ' out of 5'
+            success = False
+
+          # Fill up self._feedback.scoops[] with 1s for every scoop scooped
+
+          # For now sdoneScooping_ is set manually by trial and error
+          # Can vision confirm if the scoop has been filled? for now, No
+          # while not self.doneScooping_:
+          
+          # TODO: The scooper is on the top of the icecream box
+          # TODO: The scooper always starts at the top of the icecream box
+
+          # Use a dummy matrix for now
+          # Go to the highest point near the corner
+          # TODO: as Alona mentioned we have no idea what are the 
+          # cpablities of the hand righ now
+          # We only go horizantly (along the longest axis of the icecream box)
+          # surface = np.random.array((30, 20, 5))
+          # for i in surface.shape[0]:
+          #   for j in surface.shape[1]:
+          #     print(surface[i][j])
+     
+        #   # call scooping service with the points
+        #   #
+
+    # Send result
+    if not success:
+      self._result.success = False
+      self._result.error_message = self.scooping_human_status_
+      self.server_.set_succeeded(self._result)
+
+  # This function is called every _feedback_delay seconds
```

Comment:
I think we should not send this with a fixed rate. We should only send it when something changed, e.g. when the state-machine observing the /scooping_planning/status topic transitions to a new state.

Reply:
I prefer a fixed rate feedback :D